frontend/src/features/admin-form/settings/SettingsPage.tsx

```diff
@@ -46,17 +58,64 @@ export const SettingsPage = (): JSX.Element => {
       navigate(`${ADMINFORM_ROUTE}/${formId}/${ADMINFORM_RESULTS_SUBROUTE}`)
   }, [formId, hasEditAccess, isCollabLoading, navigate])
 
+  const tabConfig = useMemo(() => {
+    const baseConfig: TabEntry[] = [
+      {
+        label: 'General',
+        icon: BiCog,
+        component: SettingsGeneralPage,
+        path: 'general',
+      },
+      {
+        label: 'Singpass',
+        icon: BiKey,
+        component: SettingsAuthPage,
+        path: 'singpass',
+      },
+      {
+        label: 'Twilio credentials',
+        icon: BiMessage,
+        component: SettingsTwilioPage,
+        path: 'twilio-credentials',
+      },
+      {
+        label: 'Webhooks',
+        icon: BiCodeBlock,
+        component: SettingsWebhooksPage,
+        path: 'webhooks',
+      },
+      {
+        label: 'Payments',
+        icon: BiDollar,
+        component: SettingsPaymentsPage,
+        path: 'payments',
+      },
+    ]
+
+    if (
+      settings?.responseMode === FormResponseMode.Encrypt ||
+      settings?.responseMode === FormResponseMode.Email
+    ) {
+      // Add the Email Notifications page into the tab list, after singpass
+      baseConfig.splice(2, 0, {
```

Comment:
Would prefer to avoid updating baseConfig as it may grow to become a frankenstein. 

```typescript
import { isNonEmpty } from '~shared/utils/isNonEmpty'

export const SettingsPage = (): JSX.Element => {
  ...
  const tabConfig = useMemo(() => {
    const emailNotificationTab =
      settings?.responseMode === FormResponseMode.Encrypt ||
      settings?.responseMode === FormResponseMode.Email
        ? {
            label: 'Email notifications',
            icon: BiMailSend,
            component: SettingsEmailsPage,
            path: 'email-notifications',
          }
        : null

    const baseConfig: (TabEntry | null)[] = [
      {
        label: 'General',
        icon: BiCog,
        component: SettingsGeneralPage,
        path: 'general',
      },
      {
        label: 'Singpass',
        icon: BiKey,
        component: SettingsAuthPage,
        path: 'singpass',
      },
      emailNotificationTab, // Add the Email Notifications page into the tab list, after singpass
      {
        label: 'Twilio credentials',
        icon: BiMessage,
        component: SettingsTwilioPage,
        path: 'twilio-credentials',
      },
      {
        label: 'Webhooks',
        icon: BiCodeBlock,
        component: SettingsWebhooksPage,
        path: 'webhooks',
      },
      {
        label: 'Payments',
        icon: BiDollar,
        component: SettingsPaymentsPage,
        path: 'payments',
      },
    ]

    return baseConfig.filter(isNonEmpty)
  }, [settings])
  ...
}
  ```

Reply:
That makes sense! Relying on the index to insert the tab is janky indeed, thanks!